File: test_data/code/python/1668-maximum-repeating-substring.py
Python
class Solution:
    def maxRepeating(self, sequence: str, word: str) -> int:
        max_k = 0
        for k in range(1, len(sequence) // len(word) + 1):
            if sequence.replace(word * k, '', 1) == '':
                max_k = k
        return max_k

File: test_data/code/java/1668-maximum-repeating-substring.java
class Solution {
    public int maxRepeating(String sequence, String word) {
        int max = 0;
        for (int i = 0; i <= sequence.length() - word.length(); i++) {
            String temp = sequence.substring(i, i + word.length());
            if (temp.equals(word)) {
                int k = 1;
                int j = i + word.length();
                while (j < sequence.length() && sequence.substring(j - word.length(), j).equals(word)) {
                    j += word.length();
                    k++;
                }
                max = Math.max(max, k);
            }
        }
        return max;
    }
}